codelabs/custom_toolbox/custom_toolbox.md

```diff
@@ -0,0 +1,493 @@
+```xml
+<category name="Logic" categorystyle="logic_category">
+```
+to be:
+```xml
+<category css-icon="customIcon fa fa-cog" name="Logic" categorystyle="logic_category">
+```
+All the classes used to create a category can be set similar to how we set the
+icon class above. See the [Blockly toolbox documentation](https://developers.google.com/blockly/guides/configure/web/toolbox#category_css) for more information.
+
+### Add some CSS
+If you open `playground.html` you will notice that the gear icon is positioned
+incorrectly and is a bit difficult to see. We will use the `customIcon` class to
+position the icon above the label and change the color.
+
+In your `toolbox_style.css` file add:
+```css
+/* Changes color of the icon to white. */
+.customIcon {
+  color: white;
+}
+/* Stacks the icon on top of the label. */
+.blocklyTreeRowContentContainer {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+}
+.blocklyTreeRow {
+  height: initial;
+}
+```
+
+### Update setSelected
+If you open `playground.html` and click on the "Logic" category you will notice
+that the white icon now blends into the white background.
+
+In order to fix this, we are going to update our `setSelected` method to set the
+color of the icon to the color of the category when it is selected.
+
+Inside `custom_category.js` add the below line to `setSelected` if the category
+has been selected:
+
+```js
+this.iconDom_.style.color = this.colour_;
+```
+
+Add the below line to `setSelected` if the category has not been selected:
+```js
+this.iconDom_.style.color = 'white';
+```
+
+Your `setSelected` method should look similar to below:
+```js
+  /** @override */
+  setSelected(isSelected){
+     // We do not store the label span on the category, so use getElementsByClassName.
+     var labelDom = this.rowDiv_.getElementsByClassName('blocklyTreeLabel')[0];
+     if (isSelected) {
+       // Change the background color of the div to white.
+       this.rowDiv_.style.backgroundColor = 'white';
+       // Set the colour of the text to the colour of the category.
+       labelDom.style.color = this.colour_;
+       this.iconDom_.style.color = this.colour_;
+     } else {
+       // Set the background back to the original colour.
+       this.rowDiv_.style.backgroundColor = this.colour_;
+       // Set the text back to white.
+       labelDom.style.color = 'white';
+       this.iconDom_.style.color = 'white';
+     }
+     // This is used for accessibility purposes.
+     Blockly.utils.aria.setState(/** @type {!Element} */ (this.htmlDiv_),
+         Blockly.utils.aria.State.SELECTED, isSelected);
+  }
+```
+### The result
+If you open your playground, you should see a white gear above your "Logic"
+label, and it should change to blue when the category has been selected.
+
+![](./category_gear.png)
+
+![](./category_gear_selected.png)
+
+## Change the category HTML
+Currently, the element for the icon on the category is a `<span>` element. For a
```

Comment:
I was suggesting a reordering of the pages to have
1. how to change css classes (the icon part)
2. how to override methods (color border part and icon img part)
3. how to add custom items (the rest of the codelab stays where you have it)

but I also think the current organization is fine and it makes sense with the way you introduce the changes to setSelected. because i think even in the css part you need to change setSelected for the final product to look right, so you have to introduce that concept anyway, and you've already done that so everything makes sense.

so tldr: you're good, ignore me :)